cpp/run3ScoutingLooper.C

```diff
@@ -0,0 +1,587 @@
+  std::vector<float> PFRelIsoChg, PFRelIsoAll;
+  std::vector<float> mindr, maxdr;
+  std::vector<float> mindrPF, mindrJet;
+  std::vector<TLorentzVector> vec;
+  std::vector<TString> type; // types = ["G & T","G & !T","!G & T","SA & !G & !T","!SA & !G & !T"]
+
+  void clear() {
+  vtxIdxs.clear();
+  saHits.clear(); saMatchedStats.clear();
+  muHits.clear(); muChambs.clear(); muCSCDT.clear(); muMatch.clear(); muMatchedStats.clear(); muExpMatchedStats.clear(); muMatchedRPC.clear();
+  pixHits.clear(); stripHits.clear();
+  pixLayers.clear(); trkLayers.clear();
+  ch.clear();
+  pt.clear(); eta.clear(); phi.clear();
+  chi2ndof.clear();
+  ecalIso.clear(); hcalIso.clear(); trackIso.clear();
+  ecalRelIso.clear(); hcalRelIso.clear(); trackRelIso.clear();
+  dxy.clear(); dxye.clear(); dz.clear(); dze.clear();
+  dxysig.clear(); dzsig.clear();
+  PFIsoChg.clear(); PFIsoAll.clear();
+  PFRelIsoChg.clear(); PFRelIsoAll.clear();
+  mindr.clear(); maxdr.clear();
+  mindrPF.clear(); mindrJet.clear();
+  vec.clear();
+  type.clear();
+  }
+};
+
+
+std::vector<std::string> selL1Seeds = {"L1_DoubleMu_15_7",
+                                       "L1_DoubleMu4p5er2p0_SQ_OS_Mass_Min7",
+                                       "L1_DoubleMu4p5er2p0_SQ_OS_Mass_7to18",
+                                       "L1_DoubleMu4_SQ_OS_dR_Max1p2",
+                                       "L1_DoubleMu4p5_SQ_OS_dR_Max1p2"};
+
+
+void run3ScoutingLooper(std::vector<TString> inputFiles, TString year, TString process, const char* outdir="temp_data", TString label="") {
+  // Output folders and files
+  fs::create_directory(outdir);
+  fs::permissions(outdir,fs::perms::owner_all | fs::perms::group_read | fs::perms::group_exec | fs::perms::others_read | fs::perms::others_exec);
+  TFile* fout = new TFile(TString(outdir)+"/output_"+process+"_"+year+label+".root", "RECREATE");
+  TTree* tout = new TTree("tout","Run3ScoutingTree");
+
+  // Branch variables
+  float PV_x, PV_y, PV_z;
+  SV SVs;
+  int nMuonAssoc;
+  Muon Muons;
+
+  // Branch definition
+  tout->Branch("PV_x", &PV_x);
+  tout->Branch("PV_y", &PV_y);
+  tout->Branch("PV_z", &PV_z);
+
+  tout->Branch("SV_index", &SVs.index);
+  tout->Branch("SV_ndof", &SVs.ndof);
+  tout->Branch("SV_x", &SVs.x);
+  tout->Branch("SV_y", &SVs.y);
+  tout->Branch("SV_z", &SVs.z);
+  tout->Branch("SV_xe", &SVs.xe);
+  tout->Branch("SV_ye", &SVs.ye);
+  tout->Branch("SV_ze", &SVs.ze);
+  tout->Branch("SV_chi2", &SVs.chi2);
+  tout->Branch("SV_prob", &SVs.prob);
+  tout->Branch("SV_chi2OverNdof", &SVs.chi2Ndof);
+  tout->Branch("SV_lxy", &SVs.lxy);
+  tout->Branch("SV_l3d", &SVs.l3d);
+  tout->Branch("SV_mindx", &SVs.mindx);
+  tout->Branch("SV_mindy", &SVs.mindy);
+  tout->Branch("SV_mindz", &SVs.mindz);
+  tout->Branch("SV_maxdx", &SVs.maxdx);
+  tout->Branch("SV_maxdy", &SVs.maxdy);
+  tout->Branch("SV_maxdz", &SVs.maxdz);
+  tout->Branch("SV_isValid", &SVs.isValid);
+
+  tout->Branch("nMuonAssoc", &nMuonAssoc);
```

Comment:
It may be worth adding "nMuonSel" (or similar), to count muons that also pass kinematic cuts in addition to being associated with a vertex?

Reply:
The number of selected muons (or any other object) can be inferred by the size of the vector in the tree (`vec.size()` in C++ or `Length$(vec)` in TTrees), that's why I didn't included it separately. If that's not convenient/complete, I can include an extra branch with the number.